woocommerce/woocommerce-android
# Order create edit shipping lines support

Closes: #11393
Closes: #11395

### Description
This PR introduces a significant enhancement by adding support for multiple shipping lines to the order create/edit flows. This is achieved by incorporating a new shipping section into the order create/edit screen. To display the shipping details, we utilize the current shipping lines field within the order, fetching the shipping method title from the `GetShippingMethodsWithOtherValue` use case. If the method title is not present in the local DB, it is simply not displayed. 
In addition to displaying shipping lines, I cleaned up all legacy codes of the button interaction for the shipping section in the order totals sheet and the logic to display a warning when there was more than one shipping line.
Finally, I made a slight adjustment to rely on the shipping lines itemId field for the add/update/delete functions so that we can manage the order's shipping lines.

### Testing instructions

TC1
1. Open the app
2. Tap on Add a new order
3. Add a product to enable the shipping section
4. Tap on add shipping and add a new shipping line
5. Check that the new shipping section is displayed
6. Optionally, you can remove the shipping line just added to check that when there is no shipping line, the shipping section is hidden.
7. Tap on the plus button in the shipping section
8. Add a shipping line line
9. Check that the new shipping line is added to the order
10. Edit one of the shipping lines
11. Check that the changes are persisted
12. Delete one of the shipping lines
13. Check that the shipping line is removed from the order
14. Create the order

TC2
1. Using the order form TC1
2. Tap on Edit to enter the edit flow
3. Tap on the plus button in the shipping section
4. Add a shipping line line
5. Check that the new shipping line is added to the order
6. Edit one of the shipping lines
7. Check that the changes are persisted
8. Delete one of the shipping lines
9. Check that the shipping line is removed from the order

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/a3ec425a-7d58-4993-a695-8d00624f20aa
